### PR TITLE
feature: add misspell tool to check English words

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

I add tool [misspell](https://github.com/client9/misspell) to check incorrect English words spelling in this project.

I combined this feature in CI with test type of markdownlint.

In addition I use a new Dockerfile to construct new image `allencloud/mdlmisspell:v0.1`:

```
FROM ubuntu:16.04

RUN apt-get update \
    && apt-get install -y rubygems git curl \
    && gem install rake \
    && gem install bundler \
    && apt-get clean

RUN git clone https://github.com/markdownlint/markdownlint.git \
    && cd markdownlint && git checkout v0.4.0 && rake install

RUN git clone https://github.com/client9/misspell.git \
    && cd misspell && curl -L -o ./install-misspell.sh https://git.io/misspell && sh ./install-misspell.sh

RUN mv misspell/bin/misspell /usr/local/bin/
```

### Ⅱ. Does this pull request fix one issue?

none

### Ⅳ. Describe how to verify it
This addition would checkout one typo.

### Ⅴ. Special notes for reviews
none

### CODE REVIEW

**Comment by allencloud** [comment]
In this test, we have already found:

```
misspell -error ./**/* | grep -v extra | grep -v vendor
pkg/errtypes/errors.go:32:51: "useing" is a misspelling of "using"
Exited with code 2
```

So it turns out that misspell works very well.

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,29 +3,19 @@
...
       - checkout
-
-      # specify any bash command here prefixed with `run: `
       - run:
           name: use markdownlint v0.4.0 to lint markdown file (https://github.com/markdownlint/markdownlint)
           command: |
             find  ./ -name  "*.md" | grep -v vendor | grep -v extra |  grep -v commandline |  grep -v .github |  grep -v swagger |  grep -v api |  xargs mdl -r ~MD010,~MD013,~MD024,~MD029,~MD033,~MD036
+      - run:
+          name: use opensource tool client9/misspell to correct commonly misspelled English words 
+          command: find  ./ -name  "*" | grep -v extra | grep -v vendor
```
Comment (zhuangqh) [inline .circleci/config.yml]: @allencloud Forgot to execute `misspell`? `xargs misspell -error`

Comment (allencloud) [reply .circleci/config.yml]: Ah, Thanks goodness. I am afraid I made a mistake in git commit.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**